File: rename_branding.py
import os

root_dir = r"C:\Users\user\Desktop\MY RESUMES\nexus-os"
exclude_dirs = {'.git', 'node_modules', '.next', 'build', '__pycache__', 'venv', '.env', '.venv', '.gemini'}

# Be careful with order, replace longer strings first to avoid partial replacements
ordered_replacements = [
    ("NEXUS OS", "NEXUS OS"),
    ("NEXUS OS", "NEXUS OS"),
    ("nexus os", "nexus os"),
    ("nexus-os", "nexus-os"),
    ("nexus-os", "nexus-os"),
    ("NEXUS OS", "NEXUS OS"),
    ("NEXUS", "NEXUS"),
    ("nexus", "nexus")
]

def process_file(filepath):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
    except UnicodeDecodeError:
        # Skip binary files or non-utf8
        return
    except Exception as e:
        print(f"Error reading {filepath}: {e}")
        return

    new_content = content
    for old_str, new_str in ordered_replacements:
        new_content = new_content.replace(old_str, new_str)
        
    if new_content != content:
        try:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(new_content)
            print(f"Updated: {filepath}")
        except Exception as e:
            print(f"Error writing {filepath}: {e}")

for dirpath, dirnames, filenames in os.walk(root_dir):
    # Skip excluded directories
    dirnames[:] = [d for d in dirnames if d not in exclude_dirs]
    
    for filename in filenames:
        if filename.endswith(('.pyc', '.png', '.jpg', '.jpeg', '.gif', '.mp4', '.pdf', '.zip', '.tar', '.gz')):
            continue # skip media
        
        filepath = os.path.join(dirpath, filename)
        process_file(filepath)

print("Renaming complete.")

File: rename_branding.js
const fs = require('fs');
const path = require('path');

const rootDir = "C:\\Users\\user\\Desktop\\MY RESUMES\\nexus-os";
const excludeDirs = new Set(['.git', 'node_modules', '.next', 'build', '__pycache__', 'venv', '.env', '.venv', '.gemini']);

const orderedReplacements = [
    ["NEXUS OS", "NEXUS OS"],
    ["NEXUS OS", "NEXUS OS"],
    ["nexus os", "nexus os"],
    ["nexus-os", "nexus-os"],
    ["nexus-os", "nexus-os"],
    ["NEXUS OS", "NEXUS OS"],
    ["NEXUS", "NEXUS"],
    ["nexus", "nexus"]
];

function processFile(filepath) {
    try {
        let content = fs.readFileSync(filepath, 'utf8');
        let newContent = content;
        
        for (const [oldStr, newStr] of orderedReplacements) {
            newContent = newContent.split(oldStr).join(newStr);
        }
        
        if (newContent !== content) {
            fs.writeFileSync(filepath, newContent, 'utf8');
            console.log(`Updated: ${filepath}`);
        }
    } catch (e) {
        // console.error(`Skipping ${filepath}: non-utf8 or error`);
    }
}

function walkSync(dir, filelist = []) {
    let files;
    try {
        files = fs.readdirSync(dir);
    } catch(e) { return filelist; }
    
    files.forEach(function(file) {
        if (excludeDirs.has(file)) return;
        const filepath = path.join(dir, file);
        
        let stat;
        try { stat = fs.statSync(filepath); } catch(e) { return; }
        
        if (stat.isDirectory()) {
            filelist = walkSync(filepath, filelist);
        } else {
            if (!filepath.match(/\.(pyc|png|jpg|jpeg|gif|mp4|pdf|zip|tar|gz)$/)) {
                processFile(filepath);
            }
        }
    });
    return filelist;
}

walkSync(rootDir);
console.log("Renaming complete.");
